### Домашняя работа №1

##### Задание №1

Из колоды в 52 карты извлекаются случайным образом 4 карты. 

a) Найти вероятность того, что все карты – крести. 
б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

Решение:

а) Крестовых карт в колоде:
    
    n = 52/4 = 13 

соответственно вытащить одну карту крестовой масти:

    P = m/n = 13/52
    
соответствено вытащить 4 карты подряд крестовой масти:

    P = 13/52 * 12/51 * 11/50 * 10/49 = 0,25 * 0,23 * 0,22 * 0,20 = 0.00253

б) Вероятность вытащить среди 4-х карт один туз:

In [1]:
from math import factorial

In [2]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [3]:
c1 = combinations(4, 1)
c1

4

In [4]:
c2 = combinations(48, 3)
c2

17296

In [5]:
C1_sum = c1 * c2
C1_sum

69184

In [6]:
C2_sum = combinations(52, 4)
C2_sum

270725

In [7]:
# Р = число исходов благоприятствующих нашему событию / общее число исходов

P = C1_sum / C2_sum
P

0.25555083571890297

Ответ: а) вероятность вытащить все 4-ре карты крестовой масти Р = 0.00253
       б) вероятность того, что среди этих карт окажется один туз Р = 0.25555

##### Задание №2

На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. 
Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, 
откроет дверь с первой попытки?

In [8]:
c = combinations(10, 3)
P = 1 / c
P

0.008333333333333333

Ответ: вероятность того, что человек не знающий код откроет с первой попытки Р = 0.00833

##### Задание №3

В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. 
Какова вероятность того, что все извлеченные детали окрашены?

In [9]:
P = 9/15 * 8/14 * 7/13
P

0.1846153846153846

Ответ: вероятность того, что все извлеченные детали окрашены Р = 0.18461

##### Задание №4

В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

In [10]:
c = combinations(100, 2)
P = 1 / c
P

0.00020202020202020202

Ответ: вероятность того, что 2 приобретенных билета окажутся выигрышными Р = 0.000202